### ISSUE TITLE
Add convenience attributes for generating namespace packages

### ISSUE THREAD

**Issue by petri** [audreyr/cookiecutter-pypackage#8]
Namespaces can be implemented in cookiecutter templates by appropriately parsing the repo_name within the jinja-templated directory names.

However it would be more readable and convenient if cookiecutter itself parsed the repo_name into levels and put them into the "cookiecutter" namespace that's accessible from the jinja templates. So that instead of having to resort to this kind of directory structure:

```jinja
{{cookiecutter.repo_name.split('.')[0]}} / {{cookiecutter.repo_name.split('.')[1]}} 
```
...we could have a nice structure like this:

```jinja
{{cookiecutter.package} / {{cookiecutter.subpackage}
```
... and likewise for the third level.

**Comment by petri** [comment]
Is the practice to close down ideas that won't be implemented? It would be nice to at least label such issues "needs implementer" or something.

**Comment by pydanny** [comment]
@petri While your idea was interesting, it's not something that in our belief fit the design of having a `{{cookiecutter}}` object that contained all the data. Combining variables and directory names in this way could cause for challenging-to-debug problems in templates, and IMO isn't intuitive at the template level. Hence why I believe it was never implemented. This should have been vocalized earlier, by myself (or Audrey if it was her opinion as well), and for that you have my apology.

Unfortunately, should we decide to implement this design change now (which we won't), it would cause backwards-breaking changes to [literally hundreds of cookiecutter templates](https://github.com/search?q=cookiecutter&ref=opensearch). We simply can't cause this kind of damage to the Cookiecutter community.

By closing this issue we are stating that it's a design consideration that didn't make it into Cookiecutter. We won't accept a pull request for it, hence why a "needs implementer" label wouldn't make sense for it.

All this said, don't let this experience stop you from making suggestions. This was a very creative one, and well worth thinking about. We appreciate your thoughts and look forward to more of them in the future.

**Comment by audreyr** [comment]
@petri By default please assume ignorance not malice 😢 I closed this because it was old and stale, no PRs had been submitted for it by you or anyone else, and the wording was unclear but I didn't want to annoy you by asking you for an update, PR, or clarification years later. I also was worried that it might confuse sprinters during our PyCon US sprint. Finally, without any PRs submitted, updates, or new comments, I assumed lack of post-2013 interest.
 
Let me ask you for clarification now:
* Are you referring to a change in Cookiecutter core or in pypackage?
* Can you provide a more concrete example of how this feature would work?
* If you mean a change in cookiecutter-pypackage, would you have the time to implement it, complete with tests, docs, and ensuring it works cross-platform?

**Comment by petri** [comment]
No offence taken, I was just asking for clarification since I had no idea why it was closed. That was all.

After reading your thoughtful explanations (thank you!) and looking at this again, it seems to me that I should've submitted this issue to the cookiecutter core rather than here. Apologies for that.

It's been a while, but I think the idea was just to provide a convenience for generating nested packages in cases where the package structure can be deduced from the package name given to cookiecutter, such as with Python namespace packages.

To be honest, I did not think the idea through that much back then, in terms of how applicable this would be for various languages that cookiecutter is used for;  ie. how prevalently they use package name conventions that sufficiently often correspond with particular package directory structures etc. I suspect that applicable conventions might well be used commonly enough to satisfy the 80/20 rule though.

At the time, I would've been happy to help with the implementation, but nowadays I'm way too overloaded for that.
